### ISSUE TITLE
zstd CLI provides no output when it can't create destination file

### ISSUE THREAD

**Issue by chipturner** [Cyan4973/zstd#231]
When trying to compress a file that a user doesn't have permission to, or on a read-only filesystem, or possibly other cases, zstd fails and correctly returns 1 but provides no diagnostics.  Example as non-root user:

```shell
$ gzip /etc/passwd; echo $?
gzip: /etc/passwd.gz: Permission denied
1
$ zstd /etc/passwd; echo $?
1
```

**Comment by Cyan4973** [comment]
Possibly related to #232 , which also detects something and exits silently.

**Comment by Cyan4973** [comment]
Latest update in "dev" branch will no answer : 

```
$ zstd /etc/passwd; echo $?
zstd: /etc/passwd.zst: cannot open 
1
```

It's not the same error message as spelled by `gzip` though, 
it just understands it cannot create destination file, it doesn't analyze further.

**Comment by chipturner** [comment]
Would it be possible to get the error message from the OS?  This is an improvement but that would be ideal.

**Comment by Cyan4973** [comment]
That's a good idea. But how to do that ?

**Comment by chipturner** [comment]
It depends on the API you're using; it would typically be from `strerror` after the open/fopen fails

**Comment by Cyan4973** [comment]
Indeed, it works !
```
./zstd /etc/passwd; echo $?
zstd: /etc/passwd.zst: Permission denied 
1
```
Thanks for the hint !

**Comment by Cyan4973** [comment]
Fixed in v0.7.2
